### DISCUSSION TITLE
incorrect hydrostatic pressure

### DISCUSSION THREAD

**Discussion by nlosacco** [idaholab/moose#27118] [Q&A Modules: Porous Flow]
Hi all,

just trying to get the hydrostatic pwp distribution in a 10 m soil column. I set gravity = 10. and density = 1000.
Why am I getting the wrong pwp at the bottom? It should be u = 1.E+5, while I get u=9.392233e+04. Can someone tell me what I'm missing here? Please see the attached input:

```
[Mesh]
  [soil]
   type = GeneratedMeshGenerator
    dim = 2
    elem_type = QUAD4
    nx = 4
    ny = 20
    xmax = 2.
    ymax = 10.
  []
[]

[GlobalParams]
  PorousFlowDictator = dictator
  gravity = '0. -10. 0.'
[]

[Variables]
  [u]
    family = LAGRANGE
    order = FIRST
  []
[]

[PorousFlowBasicTHM]
  porepressure = u
  coupling_type = Hydro
  fp = the_simple_fluid
[]

[FluidProperties]
  [the_simple_fluid]
    type = SimpleFluidProperties
    bulk_modulus = 2E9
    viscosity = .001
    density0 = 1000.
  []
[]
[Materials]
  [porosity]
    type = PorousFlowPorosity
    porosity_zero = 0.1
  []
  [permeability_aquifer]
    type = PorousFlowPermeabilityConst
    permeability = '1.E-3 0 0   0 1.E-3 0   0 0 1.E-3'
  []
[]

[BCs]
  [top]
     type = DirichletBC
     variable = u
     value = 0.
     boundary = top
  []
[]

[Postprocessors]
   [aaa]
     type = PointValue
     point = '0. 0. 0.'
     variable = u
   []
[]

[Executioner]
   type = Steady
   solve_type = Newton
[]
```

**Comment by GiudGiud** [comment]
Hello

Is this result converged spatially (mesh-wise)? and nonlinear-solve wise?
what does the convergence log look like?

Guillaume

**Reply by nlosacco** [reply]
Yeah, I mean... the pwp profile with depth should have slope 1.E+4 and instead it has slope 0.93E+4. This is consistently observed in much more complex domains/meshes. If I use a `PorousFlowFullySaturatedDarcyBase`, instead of `PorousFlowBasicTHM`, the pwp is predicted correctly!
But then, when I move to a complex Transient analysis (simulation of a falling head test), the analysis does not converge (while it does converge with `PorousFlowBasicTHM`

**Reply by GiudGiud** [reply]
@cpgr will know the differences in implementation of these two approaches.
I would say try to refine the mesh and make sure the nonlinear solve is converged for now.

**Reply by nlosacco** [reply]
mesh refined (10*n elms), nl_abs_tol = 1.E-20, tried both first and second order elements.
Result does not change at all

**Reply by GiudGiud** [reply]
ok good to know, thanks for checking

If you want to investigate this yourself you can use Debug/show_actions and Debug/show_execution_order=ALWAYS to see how the simulation with `PorousFlowBasicTHM` differs from the one with `PorousFlowFullySaturatedDarcyBase`

**Reply by nlosacco** [reply]
thanks for the suggestion, I'll certainly do, for sake of curiosity.
In any case, setting `temperature = 0` (whatever the physical meaning is) for `PorousFlowBasicTHM` (default is 293), yields the correct value. Hence, I guess there are some hidden dependencies on temperature... maybe I missed it, but I think this should be clarified in the docs

**Reply by 1runer** [reply]
Hi @nlosacco 

Some comments:
FluidProperties in MOOSE are never constant, so e.g. you have to consider the change in density as function of pressure and temperature. You can modify the fluid compressibility to make the fluid stiffer. You can output the fluid density as AuxVariable to check the density in your simulation.

[PorousFlowTHM ](https://mooseframework.inl.gov/source/actions/PorousFlowBasicTHM.html )action uses [PorousFlowFullySaturatedMassTimeDerivative](https://mooseframework.inl.gov/source/kernels/PorousFlowFullySaturatedMassTimeDerivative.html) and assumes pressure as variable to solve. Furthermore standard in the action is that the mass balance is NOT multiplied by density (on both sides) which makes the equation linear and quasi-static in each timestep.
On the other hand if you use [PorousFlowMassTimeDerivative](https://mooseframework.inl.gov/source/kernels/PorousFlowMassTimeDerivative.html), e.g. by using the [PorousFlowUnsaturated Action](https://mooseframework.inl.gov/source/actions/PorousFlowUnsaturated.html), the equations solve for the density change and you have a "real" transient simulation. 


EDIT:
The best to fit you results might be to use PorousFlowTHM action and change the FluidProperties to a high bulk modulus (stiff water) and thermal_expansion to 0.

**Reply by nlosacco** [reply]
Hi @1runer 

I already tried and set a very high bulk modulus (about 1E3 GPa) and thermal_expansion = 0, but yet the result didn't change, unless temperature=0

**Reply by cpgr** [reply]
@1runer is certainly correct here - the fluid density isn't 1000 kg/m3 in your case. If you set thermal_expansion = 0 and re-run it, it should give you the expected result (I know you said that it didn't but I know it will).
